core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1035,23 +1088,27 @@ private void copyTable(SqlExecutionContext executionContext, CopyModel model) th
     private TableWriter copyTableData(CharSequence tableName, RecordCursor cursor, RecordMetadata cursorMetadata) {
         TableWriter writer = new TableWriter(configuration, tableName, messageBus, false, DefaultLifecycleManager.INSTANCE);
         try {
-            RecordMetadata writerMetadata = writer.getMetadata();
-            entityColumnFilter.of(writerMetadata.getColumnCount());
-            RecordToRowCopier recordToRowCopier = assembleRecordToRowCopier(asm, cursorMetadata, writerMetadata, entityColumnFilter);
-
-            int timestampIndex = writerMetadata.getTimestampIndex();
-            if (timestampIndex == -1) {
-                copyUnordered(cursor, writer, recordToRowCopier);
-            } else {
-                copyOrdered(writer, cursor, recordToRowCopier, timestampIndex);
-            }
+            copyTableData(cursor, cursorMetadata, writer);
             return writer;
         } catch (CairoException e) {
             writer.close();
             throw e;
         }
     }
 
+	public void copyTableData(RecordCursor cursor, RecordMetadata cursorMetadata, TableWriter writer) {
+		RecordMetadata writerMetadata = writer.getMetadata();
+		entityColumnFilter.of(writerMetadata.getColumnCount());
+		RecordToRowCopier recordToRowCopier = assembleRecordToRowCopier(asm, cursorMetadata, writerMetadata, entityColumnFilter);
```

Comment:
If backup is a regular thing, which it might be, it would make sense to cache copiers. Copier can be cached against table name + structure version.